sets/set.go

```diff
@@ -20,6 +20,28 @@ func NewSet[T comparable](vs ...T) Set[T] {
 	return s
 }
 
+// Create a set from a slice, applying a mapping function first to
+// translate to a comparable type.
+func FromSlice[T1 any, T2 comparable](vs []T1, f func(T1) T2) Set[T2] {
```

Comment:
Because this operates on a slice, I'd put it in the `slices` package, maybe in `slices/map.go` and call it `MapAsSet` or something like that.